scalac-scoverage-runtime/js/src/main/scala/scalajssupport/File.scala

```diff
@@ -71,6 +71,14 @@ object File {
       RhinoFile
     else if (globalObject.hasOwnProperty("callPhantom").asInstanceOf[Boolean])
       PhantomFile
+    else if (
+      globalObject
+        .hasOwnProperty("navigator")
+        .asInstanceOf[Boolean] && globalObject.navigator.userAgent
+        .asInstanceOf[js.UndefOr[String]]
+        .exists(_.contains("jsdom"))
+    )
+      JSDOMFile
```

Comment:
Adapted from https://github.com/scala-js/scala-js-macrotask-executor/pull/17. See the `userAgent` here: https://github.com/jsdom/jsdom#advanced-configuration